packages/signers/src/web3auth/__tests__/signer.test.ts

```diff
@@ -0,0 +1,125 @@
+import * as AASignersModule from "@alchemy/aa-signers";
+import type { RequestArguments } from "@web3auth/base";
+import { Web3Auth } from "@web3auth/modal";
+import { Web3AuthSigner } from "../signer.js";
+
+describe("Web3Auth Signer Tests", () => {
+  it("should correctly error when signing message if not authenticated", async () => {
+    const signer = await givenSigner(false);
+
+    const signMessage = signer.signMessage("test");
+    await expect(signMessage).rejects.toThrowErrorMatchingInlineSnapshot(
+      '"Not authenticated"'
+    );
+  });
+
+  it("should correctly authenticate", async () => {
+    const spy = vi.spyOn(AASignersModule, "Web3AuthSigner");
+    givenSigner();
+    expect(spy.mock.calls).toMatchInlineSnapshot("[]");
```

Comment:
shouldn't this vaidate something specific to authenticated state?

Reply:
looking at your other test cases, I think it'd be more valuable to add test cases for all of the methods that require authentication and validate that they correctly throw. 

Right now the tests below only test happy paths